controllers/cipool_controller_test.go

```diff
@@ -0,0 +1,59 @@
+package controllers
+
+import (
+	"context"
+	"testing"
+
+	ofcirv1 "github.com/openshift/ofcir/api/v1"
+	"github.com/openshift/ofcir/pkg/reconcilertest"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func TestCIPoolController(t *testing.T) {
+
+	cases := []struct {
+		name     string
+		testCase reconcilertest.Testable
+	}{
+		{
+			name: "finalizer is immediately added for a new pool",
+			testCase: newCIPoolScenario().
+				Setup(scenarioPoolWithSingleCir).
+				ReconcileUntil(func(client client.Client, obj *ofcirv1.CIPool) bool {
+					return obj.Status.State == ofcirv1.StatePoolAvailable && controllerutil.ContainsFinalizer(obj, ofcirv1.OfcirFinalizer)
+				}),
+		},
+		{
+			name: "finalizer is immediately added for a new pool",
+			testCase: newCIPoolScenario().
+				Setup(scenarioWithEmptyPool).
+				ReconcileUntil(func(client client.Client, obj *ofcirv1.CIPool) bool {
+					return obj.Status.State == ofcirv1.StatePoolAvailable
+				}, "start with an empty pool").
+				Then(func(t *testing.T, client client.Client, obj *ofcirv1.CIPool) {
+					assert.Equal(t, obj.Status.Size, 0)
```

Comment:
reverted arguments to assert results in confusing error. Should be  "assert.Equal(t, 0, obj.Status.Size)"

Reply:
Right, good point